src/Workspaces/Core/Portable/SymbolKey/SymbolKey.BodyLevelSymbolKey.cs

```diff
@@ -162,8 +164,42 @@ public static SymbolKeyResolution Resolve(SymbolKeyReader reader, out string? fa
                     }
                 }
 
-                failureReason = $"({nameof(BodyLevelSymbolKey)} '{name}' not found)";
+                failureReason = $"({nameof(BodyLevelSymbolKey)} '{name}' not found -> {totalFailureReason})";
                 return default;
+
+                bool TryResolveLocation(Location loc, int index, out SymbolKeyResolution resolution, out string? reason)
+                {
+                    var resolutionOpt = reader.ResolveLocation(loc);
+                    if (resolutionOpt == null)
+                    {
+                        reason = $"location {index} failed to resolve";
```

Comment:
Do you want the string to also include 'location' rather than the index? It seems like that'd be more useful and also prevents the need of passing index?